clients/fides-js/src/lib/fidesString.ts

```diff
@@ -0,0 +1,82 @@
+import { CmpApi } from "@iabgpp/cmpapi";
+
+import { FIDES_SEPARATOR } from "./tcf/constants";
+import { VendorSources } from "./tcf/vendors";
+
+/** Default GPP string to use when no CmpApi is provided or when it returns no string */
+export const DEFAULT_GPP_STRING = "DBAA";
+
+/**
+ * Decodes a Fides string into its component parts.
+ *
+ * The Fides string format is: `TC_STRING,AC_STRING,GPP_STRING` where:
+ * - TC_STRING: The TCF (Transparency & Consent Framework) string
+ * - AC_STRING: The Additional Consent string, which is derived from TC_STRING
+ * - GPP_STRING: The Global Privacy Platform string
+ *
+ * Rules:
+ * 1. If the string is empty or undefined, all parts are empty strings
+ * 2. If only one part exists, it's treated as the TC string
+ * 3. AC string can only exist if TC string exists (as it's derived from TC)
+ * 4. GPP string is independent and can exist with or without TC/AC strings
+ *
+ * @example
+ * // Complete string with all parts
+ * decodeFidesString("CPzvOIA.IAAA,1~2.3.4,DBABLA~BVAUAAAAAWA.QA")
+ * // Returns { tc: "CPzvOIA.IAAA", ac: "1~2.3.4", gpp: "DBABLA~BVAUAAAAAWA.QA" }
+ *
+ * // TC string only
+ * decodeFidesString("CPzvOIA.IAAA")
+ * // Returns { tc: "CPzvOIA.IAAA", ac: "", gpp: "" }
+ *
+ * // GPP string only (with empty TC and AC)
+ * decodeFidesString(",,DBABLA~BVAUAAAAAWA.QA")
+ * // Returns { tc: "", ac: "", gpp: "DBABLA~BVAUAAAAAWA.QA" }
+ *
+ * @param fidesString - The combined Fides string to decode
+ * @returns An object containing the decoded TC, AC, and GPP strings
+ */
+export const decodeFidesString = (fidesString: string) => {
```

Comment:
Could we add return typing and formalize this? Then we can enforce on each usage of `decodeFidesString`